woo-razorpay.php

```diff
@@ -411,13 +419,123 @@ public function autoEnableWebhook()
 
         }
 
+        public function pluginInstrumentation()
+        {
+            $trackObject = new TrackPluginInstrumentation($_POST['woocommerce_razorpay_key_id'], $_POST['woocommerce_razorpay_key_secret']);
+
+            if (empty($_POST['woocommerce_razorpay_key_id']) or
+                empty($_POST['woocommerce_razorpay_key_secret']))
+            {
+                error_log('Key Id and Key Secret are required.');
+                return;
+            }
+
+            $existingVersion = get_option('rzp_woocommerce_current_version');
+
+            if(isset($existingVersion))
+            {
+                update_option('rzp_woocommerce_current_version', get_plugin_data(__FILE__)['Version']);
+            }
+            else
+            {
+                add_option('rzp_woocommerce_current_version', get_plugin_data(__FILE__)['Version']);
+            }
+
+            try
+            {
+                $api = new Api($_POST['woocommerce_razorpay_key_id'], $_POST['woocommerce_razorpay_key_secret']);
+                $orderCount = $api->request->request('GET', 'orders')['count'];
+                $isTransactingUser = ($orderCount > 0) ? true : false;
+            }
+            catch (\Razorpay\Api\Errors\Error $e)
+            {
+                ?>
+                <div class="notice error is-dismissible" >
+                    <p><b><?php _e( 'Please check Key Id and Key Secret.'); ?></b></p>
+                </div>
+                <?php
+                error_log('Please check Key Id and Key Secret.');
+                return;
+            }
+            catch (\Exception $e)
+            {
+                ?>
+                <div class="notice error is-dismissible" >
+                    <p><b><?php _e( 'something went wrong'); ?></b></p>
+                </div>
+                <?php
+                error_log($e->getMessage());
+                return;
+            }
+
+            $authEvent = '';
+            $pluginStatusEvent = '';
+
+            $authProperties = [
+                'plugin_name'             => get_plugin_data(__FILE__)['Name'],
+                'plugin_version'          => get_plugin_data(__FILE__)['Version'],
+                'event_timestamp'         => time(),
+                'woocommerce_version'     => WOOCOMMERCE_VERSION,
```

Comment:
@abdulwahidsharief Please include all woocommerce_version within [rzpTrackSegment](https://github.com/razorpay/razorpay-woocommerce/pull/305/files#r911386538) method at class TrackPluginInstrumentation

Reply:
@enishant added as default properties for all events